theano/tensor/basic.py

```diff
@@ -4078,6 +4078,79 @@ def R_op(self, inputs, eval_points):
             return [None]
         return self.make_node(*eval_points).outputs
 
+    def c_code_cache_version(self):
+        return (1,)
+
+    def c_code(self, node, name, inputs, outputs, sub):
+        x, = inputs
+        out, = outputs
+        outdim = self.outdim
+        fail = sub['fail']
+        return """
+        if (%(outdim)s == PyArray_NDIM(%(x)s))
+        {
+            if (NULL != %(out)s && %(x)s != %(out)s)
+                Py_XDECREF(%(out)s);
+
+            %(out)s = %(x)s;
+        }
+        else
+        {
+            if (%(outdim)s == 1)
+            {
+                if (NULL != %(out)s)
+                    Py_XDECREF(%(out)s);
+
+                npy_intp size = PyArray_SIZE(%(x)s);
+                PyArray_Dims newshape;
+                newshape.ptr = &size;
+                newshape.len = 1;
+                %(out)s = (PyArrayObject*)PyArray_Newshape(%(x)s,
+                                                           &newshape,
+                                                           NPY_CORDER);
```

Comment:
Are you sure about the NPY_CORDER? I do not understand it well. I think it is fine, but I'm not 100% sure.